benches/nexmark/main.rs

```diff
@@ -173,13 +203,33 @@ fn main() -> Result<()> {
             format!("{}", r.num_events.to_formatted_string(&Locale::en)),
             String::from("1"),
             format!("{0:.3}", r.elapsed.as_secs_f32()),
-            format!("{0:.3}", r.elapsed.as_secs_f32()),
             format!(
                 "{0:.3} K/s",
                 r.num_events as f32 / r.elapsed.as_secs_f32() / 1000.0
             ),
+            format!("{0:.3}", r.usr_cpu.as_secs_f32()),
+            format!("{0:.3}", r.sys_cpu.as_secs_f32()),
+            format!("{}", r.max_rss.to_formatted_string(&Locale::en)),
         ]
     }));
 
     Ok(())
 }
+
+fn duration_for_timeval(tv: timeval) -> Duration {
+    Duration::new(tv.tv_sec as u64, tv.tv_usec as u32 * 1_000)
+}
+
+// Returns the user CPU, system CPU and maxrss (in Kb) for the current thread.
+unsafe fn rusage_thread() -> (Duration, Duration, u64) {
+    let mut ru: rusage = mem::zeroed();
```

Comment:
Use [`MaybeUninit::uninit()`](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.uninit) for this instead (then `ru.as_mut_ptr()` instead of `&mut ru`)